c7n/resources/quotas.py

```diff
@@ -0,0 +1,477 @@
+        'request-history', rinherit=ValueFilter.schema
+    )
+
+    permissions = ('servicequota:ListRequestedServiceQuotaChangeHistory',)
+
+    def get_related(self, resources):
+        resource_manager = self.get_resource_manager()
+        related_ids = self.get_related_ids(resources)
+        related = resource_manager.resources()
+        result = {}
+        for r in related:
+            result.setdefault(r[self.RelatedIdsExpression], [])
+            if r[self.RelatedIdsExpression] in related_ids:
+                result[r[self.RelatedIdsExpression]].append(r)
+        return result
+
+    def _add_annotations(self, related_ids, resource):
+        resources = self.get_related([resource])
+        a_resources = resources.get(resource[self.RelatedIdsExpression], [])
+        akey = 'c7n:%s' % self.AnnotationKey
+        resource[akey] = a_resources
+
+
+@ServiceQuota.action_registry.register('request-increase')
+class Increase(Action):
+    """
+    Request a limit increase for a service quota
+
+    .. code-block:: yaml
+
+        policies:
+          - name: request-limit-increase
+            resource: aws.service-quota
+            filters:
+              - type: value
+                key: QuotaCode
+                value: L-foo
+            actions:
+              - type: increase
+                multiplier: 1.2
+    """
+
+    schema = type_schema('request-increase', multiplier={'type': 'number', 'minimum': 1.0})
+    permissions = ('servicequota:RequestServiceQuotaIncrease',)
+
+    def process(self, resources):
+        client = local_session(self.manager.session_factory).client('service-quotas')
+        multiplier = self.data.get('multiplier', 1.2)
+        error = None
+        for r in resources:
+            count = math.floor(multiplier * r['Value'])
+            if not r['Adjustable']:
+                continue
+            try:
+                client.request_service_quota_increase(
+                    ServiceCode=r['ServiceCode'],
+                    QuotaCode=r['QuotaCode'],
+                    DesiredValue=count
+                )
+            except client.exceptions.QuotaExceededException as e:
+                error = e
+                self.log.error('Requested:%s exceeds quota limit for %s' % (count, r['QuotaCode']))
+                continue
+            except (client.exceptions.AccessDeniedException,
+                    client.exceptions.DependencyAccessDeniedException,):
+                raise PolicyExecutionError('Access Denied to increase quota: %s' % r['QuotaCode'])
+            except (client.exceptions.NoSuchResourceException,
+                    client.exceptions.InvalidResourceStateException,
+                    client.exceptions.ResourceAlreadyExistsException,) as e:
+                error = e
+                continue
+        if error:
+            raise PolicyExecutionError from error
+
+
+@ServiceQuota.action_registry.register('add-to-template')
+class AddToTemplate(Action):
```

Comment:
kind on the fence for this action, docs here: https://docs.aws.amazon.com/servicequotas/latest/userguide/organization-templates.html

Reply:
@kapilt thoughts?

Reply:
i'm also on the fence, its only viable from the organization master account afaics which is tricky to ensure outside of policye execution. if its a one time activity, its generally something we would defer on, but quota mgmt template isn't per se one time, nor is it clear that it applies to existing accounts. tbh it feels more like its own resource re templates.